src/main/java/io/vertx/pgclient/impl/codec/DataTypeCodec.java

```diff
@@ -1310,6 +1316,48 @@ private static byte decodeHexChar(byte ch) {
     return (byte)(((ch & 0x1F) + ((ch >> 6) * 0x19) - 0x10) & 0x0F);
   }
 
+  private static boolean isHexFormat(int index, int len, ByteBuf buff) {
+    return len >= 2 && buff.getByte(index) == '\\' && buff.getByte(index + 1) == 'x';
+  }
+
+  private static Buffer decodeEscapeByteaStringToBuffer(int index, int len, ByteBuf buff) {
```

Comment:
I think we should keep this consistent with `decodeHexStringToBytes` that returns a byte[], so I'm inclined to change `decodeHexStringToBytes` to return a `Buffer` as well instead of having `textDecodeBYTEA` wrap the returned `byte[]` with a `Buffer`.